molo/core/models.py

```diff
@@ -636,7 +636,7 @@ def get_main_language_page(self):
         try:
             return self.translated_pages.get(
                 language__is_main_language=True).specific
-        except self.DoesNotExist:
+        except:
```

Comment:
Isn't this a really bad idea? just catching all exceptions?
Can't we catch the expected exceptions instead?